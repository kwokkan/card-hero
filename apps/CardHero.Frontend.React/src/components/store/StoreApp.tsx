import React, { Component } from "react";
import { Store } from "./Store";

export class StoreApp extends Component<{}, {}> {
    render() {
        return (
            <div className="col-lg-12">
                <Store />
            </div>
        );
    }
}
