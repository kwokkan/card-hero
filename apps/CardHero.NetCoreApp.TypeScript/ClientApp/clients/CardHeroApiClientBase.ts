// no export as this is used by NSwag generated api clients
class CardHeroApiClientBase {
    getBaseUrl(defaultBaseUrl: string, baseUrl: string): string {
        if (!baseUrl) {
            return defaultBaseUrl;
        }

        if (baseUrl.endsWith('/')) {
            // remove trailing slash as NSwag uses '/' as part of the api url
            return baseUrl.substring(0, baseUrl.length - 2);
        }

        return baseUrl;
    }
}

module.exports = CardHeroApiClientBase;