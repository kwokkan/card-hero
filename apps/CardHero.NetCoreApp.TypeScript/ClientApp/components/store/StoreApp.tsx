import React, { Component } from "react";
import { Layout } from "../shared/Layout";
import { Store } from "./Store";

interface IStoreAppProps {
}

interface IStoreAppState {
}

export class StoreApp extends Component<IStoreAppProps, IStoreAppState> {
    constructor(props: IStoreAppProps) {
        super(props);
    }

    render() {
        return (
            <Layout>
                <Store />
            </Layout>
        );
    }
}