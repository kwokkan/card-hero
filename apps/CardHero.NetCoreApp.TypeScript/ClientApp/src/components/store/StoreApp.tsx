import React, { Component } from "react";
import { Store } from "./Store";

interface IStoreAppProps {
}

interface IStoreAppState {
}

export class StoreApp extends Component<IStoreAppProps, IStoreAppState> {
    constructor(props: IStoreAppProps) {
        super(props);
    }

    render() {
        return (
            <div className="col-lg-12">
                <Store />
            </div>
        );
    }
}
