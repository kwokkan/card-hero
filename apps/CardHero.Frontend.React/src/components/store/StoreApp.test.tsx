/*
 * @jest-environment jsdom
 */
import MockDate from "mockdate";
import renderer, { act } from "react-test-renderer";
import { StoreApiClient, StoreItemModel } from "../../clients/clients";
import { StoreApp } from "./StoreApp";

const mockDate = new Date(2019, 11, 18, 3, 4, 5, 6);

beforeEach(() => {
    MockDate.set(mockDate);
});

afterEach(() => {
    MockDate.reset();
});

test("<StoreApp /> renders correctly with no items", () => {
    const tree = renderer
        .create(
            <StoreApp />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<StoreApp /> renders correctly with items", async () => {
    StoreApiClient.prototype.get = jest.fn(() => Promise.resolve([
        new StoreItemModel({
            id: 1,
            name: "First",
            description: "First bundle",
            cost: 100,
            itemCount: 1
        }),
        new StoreItemModel({
            id: 2,
            name: "Second",
            description: "Second bundle",
            cost: 500,
            itemCount: 5,
            expiry: new Date(2019, 11, 19)
        })
    ]));

    let tree: renderer.ReactTestRenderer;

    await act(async () => {
        tree = renderer
            .create(
                <StoreApp />
            );
    });

    const json = tree.toJSON();

    expect(json).toMatchSnapshot();
});
