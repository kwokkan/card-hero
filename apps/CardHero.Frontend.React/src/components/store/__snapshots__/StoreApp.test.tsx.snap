// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreApp /> renders correctly with items 1`] = `
<div
  className="col-lg-12"
>
  <div
    className="row"
  >
    <div
      className="col-lg-4"
    >
      <div
        className="card"
      >
        <h4
          className="card-header"
        >
          <span>
            First
          </span>
          <span
            className="pull-right badge-pill badge-secondary"
          >
            1
          </span>
        </h4>
        <div
          className="card-body"
        >
          <p>
            First bundle
          </p>
        </div>
        <div
          className="card-footer"
        >
          <button
            className="btn btn-primary pull-right"
            onClick={[Function]}
            type="button"
          >
            Buy for 
            <span
              title="100"
            >
              <span
                className="coin-group"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-coins fa-w-16 fa-fw coin-silver"
                  data-icon="coins"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 405.3V448c0 35.3 86 64 192 64s192-28.7 192-64v-42.7C342.7 434.4 267.2 448 192 448S41.3 434.4 0 405.3zM320 128c106 0 192-28.7 192-64S426 0 320 0 128 28.7 128 64s86 64 192 64zM0 300.4V352c0 35.3 86 64 192 64s192-28.7 192-64v-51.6c-41.3 34-116.9 51.6-192 51.6S41.3 334.4 0 300.4zm416 11c57.3-11.1 96-31.7 96-55.4v-42.7c-23.2 16.4-57.3 27.6-96 34.5v63.6zM192 160C86 160 0 195.8 0 240s86 80 192 80 192-35.8 192-80-86-80-192-80zm219.3 56.3c60-10.8 100.7-32 100.7-56.3v-42.7c-35.5 25.1-96.5 38.6-160.7 41.8 29.5 14.3 51.2 33.5 60 57.2z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                 
                1
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-lg-4"
    >
      <div
        className="card"
      >
        <h4
          className="card-header"
        >
          <span>
            Second
          </span>
          <span
            className="pull-right badge-pill badge-secondary"
          >
            5
          </span>
        </h4>
        <div
          className="card-body"
        >
          <p>
            Second bundle
          </p>
        </div>
        <div
          className="card-footer"
        >
          <span>
            Expires 
          </span>
          <time
            dateTime="2019-12-19T00:00:00.000Z"
            title="Thu, 19 Dec 2019 00:00:00 GMT"
          >
            in 21 hours
          </time>
          <button
            className="btn btn-primary pull-right"
            onClick={[Function]}
            type="button"
          >
            Buy for 
            <span
              title="500"
            >
              <span
                className="coin-group"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-coins fa-w-16 fa-fw coin-silver"
                  data-icon="coins"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 405.3V448c0 35.3 86 64 192 64s192-28.7 192-64v-42.7C342.7 434.4 267.2 448 192 448S41.3 434.4 0 405.3zM320 128c106 0 192-28.7 192-64S426 0 320 0 128 28.7 128 64s86 64 192 64zM0 300.4V352c0 35.3 86 64 192 64s192-28.7 192-64v-51.6c-41.3 34-116.9 51.6-192 51.6S41.3 334.4 0 300.4zm416 11c57.3-11.1 96-31.7 96-55.4v-42.7c-23.2 16.4-57.3 27.6-96 34.5v63.6zM192 160C86 160 0 195.8 0 240s86 80 192 80 192-35.8 192-80-86-80-192-80zm219.3 56.3c60-10.8 100.7-32 100.7-56.3v-42.7c-35.5 25.1-96.5 38.6-160.7 41.8 29.5 14.3 51.2 33.5 60 57.2z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                 
                5
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<StoreApp /> renders correctly with no items 1`] = `
<div
  className="col-lg-12"
>
  <div
    className="row"
  />
</div>
`;
